Source/WebCore/platform/graphics/skia/SkiaRecordingResult.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if USE(COORDINATED_GRAPHICS) && USE(SKIA)
+#include "GraphicsContextSkia.h"
+
+WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_BEGIN
+#include <skia/core/SkPicture.h>
+WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_END
+
+#include <wtf/TZoneMalloc.h>
+#include <wtf/ThreadSafeRefCounted.h>
+
+class SkImage;
+
+namespace WebCore {
+
+class SkiaRecordingResult final : public ThreadSafeRefCounted<SkiaRecordingResult> {
+    WTF_MAKE_TZONE_ALLOCATED(SkiaRecordingResult);
+    WTF_MAKE_NONCOPYABLE(SkiaRecordingResult);
+public:
+    virtual ~SkiaRecordingResult();
+    static Ref<SkiaRecordingResult> create(sk_sp<SkPicture>&&, SkiaImageToFenceMap&&, const IntRect& recordRect, RenderingMode, bool contentsOpaque, float contentsScale);
+
+    void waitForFenceIfNeeded(const SkImage*);
+
+    bool hasFences() const { return !m_imageToFenceMap.isEmpty(); }
+    const sk_sp<SkPicture>& picture() const { return m_picture; }
+    IntRect recordRect() const { return m_recordRect; }
+    RenderingMode renderingMode() const { return m_renderingMode; }
```

Comment:
Is this actually used?

Reply:
It has to be -- otherwise we can end up deciding between two different RenderingModes when we call determineRenderingMode() once in record() and once in replay() -- instead we should decide once on a RenderingMode, and stick with that during replay.